diff -Naur w3c-libwww-5.3.1/Makefile.am w3c-libwww-5.3.1-p/Makefile.am
--- w3c-libwww-5.3.1/Makefile.am	Wed Jun 23 20:05:01 1999
+++ w3c-libwww-5.3.1-p/Makefile.am	Wed Sep 20 18:24:21 2000
@@ -25,8 +25,8 @@
 # is platform specific, we do not install it with the other libwww include files
 install-exec-local:
 	@$(NORMAL_INSTALL)
-	$(mkinstalldirs) $(includedir)
-	$(INSTALL_DATA) $(top_builddir)/$(CONFIG_HEADER) $(includedir)/$(CONFIG_HEADER);
+	$(mkinstalldirs) $(DESTDIR)$(includedir)
+	$(INSTALL_DATA) $(DESTDIR)$(top_builddir)/$(CONFIG_HEADER) $(DESTDIR)$(includedir)/$(CONFIG_HEADER);
 
 # This is our install script provided by Soren Sandmann
 bin_SCRIPTS = libwww-config
