diff -urNp -x '*.orig' w3c-libwww-5.4.2.org/Library/Examples/Makefile.am w3c-libwww-5.4.2/Library/Examples/Makefile.am
--- w3c-libwww-5.4.2.org/Library/Examples/Makefile.am	2017-06-24 07:51:37.000000000 +0200
+++ w3c-libwww-5.4.2/Library/Examples/Makefile.am	2021-09-29 22:22:47.701642460 +0200
@@ -28,7 +28,7 @@ LDADD = \
 	../src/libwwwutils.la \
         @LIBWWWDAV@ \
 	@LIBWWWSSL@ \
-	-lm @LIBWWWZIP@ @LIBWWWWAIS@ @LIBWWWSQL@ @LIBWWWMD5@
+	-lm @LIBWWWZIP@ @LIBWWWWAIS@ @LIBWWWSQL@
 
 AM_CPPFLAGS = \
 	-I$(srcdir)/../src \
@@ -67,8 +67,6 @@ purify: $(mget_OBJECTS) $(mget_DEPENDENC
 	../src/.libs/libwwwutils.a \
         ../src/.libs/libwwwdav.a \
 	../src/.libs/libwwwzip.a \
-	../../modules/md5/.libs/libmd5.a \
-	../../modules/expat/.libs/libexpat.a
 	$(LIBS)
 
 
diff -urNp -x '*.orig' w3c-libwww-5.4.2.org/Library/src/HTDigest.c w3c-libwww-5.4.2/Library/src/HTDigest.c
--- w3c-libwww-5.4.2.org/Library/src/HTDigest.c	2017-06-24 07:51:38.000000000 +0200
+++ w3c-libwww-5.4.2/Library/src/HTDigest.c	2021-09-29 22:22:47.701642460 +0200
@@ -29,7 +29,7 @@ PUBLIC BOOL HTDigest_init (HTDigestConte
 
     switch (digesttype) {
       case HTDaMD5:
-	  MD5Init (&context->context.md5);
+	  MD5_Init (&context->context.md5);
 	  break;
       default:
           return NO;
@@ -43,7 +43,7 @@ PUBLIC BOOL HTDigest_update (HTDigestCon
     if (context) {
 	switch (context->algorithm) {
 	case HTDaMD5:
-	    MD5Update (&context->context.md5, (unsigned char *) input,
+	    MD5_Update (&context->context.md5, (unsigned char *) input,
 		       inputLen);
 	    break;
 	default:
@@ -60,7 +60,7 @@ PUBLIC BOOL HTDigest_final (unsigned cha
     if (context) {
 	switch (context->algorithm) {
 	case HTDaMD5:
-	    MD5Final (digest, &context->context.md5);
+	    MD5_Final (&context->context.md5, digest);
 	    break;
 	default:
 	    return NO;
diff -urNp -x '*.orig' w3c-libwww-5.4.2.org/Library/src/HTDigest.h w3c-libwww-5.4.2/Library/src/HTDigest.h
--- w3c-libwww-5.4.2.org/Library/src/HTDigest.h	2017-06-24 07:52:15.000000000 +0200
+++ w3c-libwww-5.4.2/Library/src/HTDigest.h	2021-09-29 22:22:47.701642460 +0200
@@ -39,7 +39,7 @@ Library.
 /* Library include files */
 #include "WWWLib.h"
 /* add the MD algorithm header files here below */
-#include "md5.h"
+#include <openssl/md5.h>
 
 #ifdef __cplusplus
 extern "C" { 
diff -urNp -x '*.orig' w3c-libwww-5.4.2.org/Library/src/Makefile.am w3c-libwww-5.4.2/Library/src/Makefile.am
--- w3c-libwww-5.4.2.org/Library/src/Makefile.am	2021-09-29 22:22:47.624975995 +0200
+++ w3c-libwww-5.4.2/Library/src/Makefile.am	2021-09-29 22:22:47.701642460 +0200
@@ -267,11 +267,7 @@ libwwwhttp_la_SOURCES = \
 	HTPEP.c
 
 libwwwhttp_la_LIBADD = libwwwmime.la libwwwstream.la libwwwcore.la libwwwutils.la \
-	../../modules/md5/libmd5.la
-
-libwwwhttp_la_CPPFLAGS = \
-	-I/usr/local/include \
-	-I${top_srcdir}/modules/md5
+	@LIBWWWMD5@
 
 libwwwnews_la_SOURCES = \
 	WWWNews.h \
@@ -427,8 +423,6 @@ libwwwsql_la_LDFLAGS = -rpath $(libdir)
 libwwwsql_la_LIBADD = libwwwcore.la libwwwutils.la $(MYSQL_LIBS)
 
 libwwwsql_la_CPPFLAGS = \
-	-I/usr/local/include \
-	-I${top_srcdir}/modules/md5 \
 	$(MYSQL_CFLAGS)
 
 BUILT_SOURCES = \
diff -urNp -x '*.orig' w3c-libwww-5.4.2.org/Robot/src/Makefile.am w3c-libwww-5.4.2/Robot/src/Makefile.am
--- w3c-libwww-5.4.2.org/Robot/src/Makefile.am	2017-06-24 07:51:38.000000000 +0200
+++ w3c-libwww-5.4.2/Robot/src/Makefile.am	2021-09-29 22:22:47.701642460 +0200
@@ -21,7 +21,7 @@ webbot_LDADD = \
 	$(top_builddir)/Library/src/libwwwcore.la \
 	$(top_builddir)/Library/src/libwwwutils.la \
 	-lm @LIBWWWXML@ @LIBWWWZIP@ @LIBWWWWAIS@ \
-	@LIBWWWSQL@ @LIBWWWMD5@ @LIBWWWSSL@ \
+	@LIBWWWSQL@ @LIBWWWSSL@ \
 	$(MYSQL_LIBS)
 
 SUBDIRS = windows
@@ -71,8 +71,6 @@ purify: $(webbot_OBJECTS) $(webbot_DEPEN
 	$(CC) $(LDFLAGS) -o purify $(webbot_OBJECTS) \
 	-lm \
 	../../Library/src/.libs/libwww.a \
-	../../modules/md5/.libs/libmd5.a \
-	../../modules/expat/.libs/libexpat.a
 	$(LIBS)
 
 
diff -urNp -x '*.orig' w3c-libwww-5.4.2.org/configure.ac w3c-libwww-5.4.2/configure.ac
--- w3c-libwww-5.4.2.org/configure.ac	2017-06-24 08:11:09.000000000 +0200
+++ w3c-libwww-5.4.2/configure.ac	2021-09-29 22:22:47.701642460 +0200
@@ -643,15 +643,15 @@ AC_ARG_WITH(md5,
     AC_MSG_RESULT(yes)
     AC_DEFINE(HT_MD5, 1, [Define to enable MD5 for HTTP access authentication.])
     WWWMD5="libmd5.la"
-    LWWWMD5="-lmd5"
-    LIBWWWMD5='${top_builddir}/modules/md5/libmd5.la'
+    LWWWMD5="-lcrypto"
+    LIBWWWMD5='-lcrypto'
     ;;
   esac ],
   AC_MSG_RESULT(yes)
   AC_DEFINE(HT_MD5)
   WWWMD5="libmd5.la"
-  LWWWMD5="-lmd5"
-  LIBWWWMD5='${top_builddir}/modules/md5/libmd5.la'
+  LWWWMD5="-lcrypto"
+  LIBWWWMD5='-lcrypto'
 )
 AC_SUBST(WWWMD5)
 AC_SUBST(LWWWMD5)
diff -urNp -x '*.orig' w3c-libwww-5.4.2.org/modules/Makefile.am w3c-libwww-5.4.2/modules/Makefile.am
--- w3c-libwww-5.4.2.org/modules/Makefile.am	2017-06-24 07:51:38.000000000 +0200
+++ w3c-libwww-5.4.2/modules/Makefile.am	2021-09-29 22:22:47.701642460 +0200
@@ -5,5 +5,4 @@ DOCS :=	$(wildcard *.html *.gif)
 EXTRA_DIST = \
 	$(DOCS)
 
-SUBDIRS = md5
 
